### PR TITLE
Create foldable/large screen helper module

### PR DESCRIPTION
**Pull request by khalp**
To simplify compose samples code by abstracting common code

(branch is based on [test-compose-gallery branch](https://github.com/microsoft/surface-duo-compose-samples/tree/khalp/test-compose-gallery))

### CODE REVIEW

File: CompanionPane/build.gradle [hunk 1]
```diff
@@ -44,6 +44,7 @@ dependencies {
     implementation composeDependencies.activityCompose
 
     implementation microsoftDependencies.twoPaneLayout
+    implementation project(':WindowState')
```
Comment (joyl1216) [inline CompanionPane/build.gradle:47]: I think we can remove the JWM dependency for all the samples now `implementation androidxDependencies.window`.

File: ComposeGallery/build.gradle [hunk 2]
```diff
@@ -49,6 +49,7 @@ dependencies {
     implementation composeDependencies.activityCompose
 
     implementation microsoftDependencies.twoPaneLayout
+    implementation project(':WindowState')
```
Comment (joyl1216) [inline ComposeGallery/build.gradle:52]: Same for this sample. We can remove the JMW dependency.

File: ComposeGallery/src/androidTest/java/com/microsoft/device/display/samples/composegallery/PaneSynchronizationTest.kt [hunk 3]
```diff
@@ -52,8 +56,16 @@ class PaneSynchronizationTest {
         composeTestRule.setContent {
             ComposeGalleryTheme {
                 ComposeGalleryApp(
-                    foldableState = FoldableState(hasFold = true, isFoldHorizontal = false),
-                    widthSizeClass = WindowSizeClass.Medium
+                    WindowState(
+                        hasFold = true,
+                        isFoldHorizontal = false,
+                        foldBounds = Rect(),
+                        foldState = FoldingFeature.State.HALF_OPENED,
```
Comment (joyl1216) [inline ComposeGallery/src/androidTest/java/com/microsoft/device/display/samples/composegallery/PaneSynchronizationTest.kt:63]: Should we use `FoldState.HALF_OPENED` here?

File: ComposeGallery/src/androidTest/java/com/microsoft/device/display/samples/composegallery/PaneSynchronizationTest.kt [hunk 4]
```diff
@@ -93,8 +105,16 @@ class PaneSynchronizationTest {
         composeTestRule.setContent {
             ComposeGalleryTheme {
                 ComposeGalleryApp(
-                    foldableState = FoldableState(hasFold = false, isFoldHorizontal = false),
-                    widthSizeClass = WindowSizeClass.Compact
+                    WindowState(
+                        hasFold = false,
+                        isFoldHorizontal = false,
+                        foldBounds = Rect(),
+                        foldState = FoldingFeature.State.HALF_OPENED,
```
Comment (joyl1216) [inline ComposeGallery/src/androidTest/java/com/microsoft/device/display/samples/composegallery/PaneSynchronizationTest.kt:112]: Same here. Should we use `FoldState.HALF_OPENED`?

**Comment by joyl1216** [comment]
The final change and test cases look good! I think once you remove the JWM dependency from the samples, you also need to remove `FoldingFeature` code. Let's ship it! :)

**Comment by hello-smile6** [comment]
> The final change and test cases look good! I think once you remove the JWM dependency from the samples, you also need to remove `FoldingFeature` code. Let's ship it! 😄

What does that mean?

**Review by joyl1216 — APPROVED** [review]

**Comment by khalp** [comment]
> > The final change and test cases look good! I think once you remove the JWM dependency from the samples, you also need to remove `FoldingFeature` code. Let's ship it! 😄
> 
> What does that mean?

@hello-smile6 this comment was about removing unnecessary Jetpack Window Manager library dependencies and imports, and the issue was fixed in the last three commits of the PR

**Comment by hello-smile6** [comment]
> > > The final change and test cases look good! I think once you remove the JWM dependency from the samples, you also need to remove `FoldingFeature` code. Let's ship it! 😄
> > 
> > 
> > What does that mean?
> 
> @hello-smile6 this comment was about removing unnecessary Jetpack Window Manager library dependencies and imports, and the issue was fixed in the last three commits of the PR

Oh.

**Pull request merged**